src/workloads/query/CsiHeavyDiskUsage.yml

```diff
@@ -0,0 +1,162 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-execution"
+Description: |
+  This workload measures bulk insert performance against a collection named "coll" in a database
+  named "heavy_io". The collection and database must be set up before running the workload. The
+  workload itself has no expectations as far as the size or schema of the data are concerned but it
+  is only useful when being run against larger datasets to actually generate heavy disk IO.
+
+  The purpose of this workload is to compare the insert performance with columnar indexes to inserts
+  without them.
+   - "Default" indexes present - a set of indexes that are typically present on the target dataset,
+     those that would be useful for the typical query workloads. These indexes should be created
+     prior to running the workload.
+   - Only a column store index. A column store index isn't expected to realistically replace all
+     other indexes in production, but it could replace some and knowing it's relative performance
+     cost could help make that decision.
+
+  The workload consists of the following phases and actors:
+   - Measure bulk inserting with the default indexes (SamplingLoader)
+   - Drop all non-_id indexes and build the column store index (RunCommand Actor)
+   - Measure bulk inserting again (SamplingLoader)
+
+Keywords:
+- columnstore
+- analytics
+- scale
+- insert
+
+Clients:
+  Default:
+    QueryOptions:
+      # Allow for longer duration since index builds may take a while.
+      socketTimeoutMS: 3_600_000  # = 1 hour
+      connectTimeoutMS: 3_600_000
+
+GlobalDefaults:
+  # The expected database and collection names.
+  Database: &db heavy_io
+  Collection: &coll coll
+
+  LoadThreads: &loadThreads 8
+  SampleSize: &sampleSize 1000
+  Batches: &batches 50  # Per Thread
+
+  ColumnStoreIndexSpec: &ColumnStoreIndexSpec
+    key: {$**: "columnstore"}
+    name: "columnstore"
+
+  MaxPhases: &maxPhases 5
+
+  SamplingLoaderPhaseConfig: &SamplingLoaderPhaseConfig
+    Database: *db
+    Collection: *coll
+    Repeat: 1
+    InsertBatchSize: 1000
+    Batches: 50  # Per Thread
+
+ActorTemplates:
+# We want to be able to compare results from the same test between the runs in presence of different
+# indexes. For this we'll have to instantiate the same actor multiple times with a unique name.
+- TemplateName: InsertFromSample
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "Insert"}}
+    Type: SamplingLoader
+    Database: *db
+    Collection: *coll
+    SampleSize: *sampleSize
+    Pipeline: [{$set: {genny_added: true}}]
+    Threads: *loadThreads
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]  # do not run by default
+        NopInPhasesUpTo: *maxPhases
+        PhaseConfig:
+          InsertBatchSize: *sampleSize
+          Repeat: 1
+          Batches: *batches
```

Comment:
I looked into the `SamplingLoader` and apparently, it doesn't have a deterministic approach in adding more documents to a given collection. Also, this way of populating the collection hides the property of documents that are going to be indexed.

IMO, there are several dimensions that are interesting to see the performance of indexing over a collection (CSI vs non-CSI): 1) having big documents with too many columns 2) having a fixed schema vs. changing schema (for each different document), etc. IMO, right now, with the current perf test setting, it's not easy to see what's the property of source collection. Can you please consider using another actor-type, like a `LoaderPhase` (e.g., [here](https://github.com/mongodb/genny/blob/7bdf0e15a6cff84e7706a2b20398ae555f2ced90/src/workloads/scale/MatViewParamsGeneric.yml#L85-L93)) to make the document structure more clear in the experiment setup?